ragna/source_storages/_chroma.py

```diff
@@ -36,14 +43,60 @@ def __init__(self) -> None:
             )
         )
 
-    def _get_collection(self, corpus_name: str) -> chromadb.Collection:
-        return self._client.get_or_create_collection(
-            corpus_name, embedding_function=self._embedding_function
-        )
-
     def list_corpuses(self) -> list[str]:
         return [collection.name for collection in self._client.list_collections()]
 
+    def _get_collection(
+        self, corpus_name: str, *, create: bool = False
+    ) -> chromadb.Collection:
+        if create:
+            return self._client.get_or_create_collection(
+                corpus_name, embedding_function=self._embedding_function
+            )
+
+        collections = list(self._client.list_collections())
```

Comment:
Nitpick: Why is it important to distinguish between `no_corpuses_available` and `non_existing_corpus`?

If we didn't need to distinguish between them, then perhaps we could just use `self._client.get_collection`, which throws a `ValueError` if the collection (corpus) does not exist.